src/main/java/org/terasology/fluid/system/FluidAuthoritySystem.java

```diff
@@ -97,13 +99,19 @@ public void fillFluidContainerItem(ActivateEvent event, EntityRef item, FluidCon
         if (fluidContainer.fluidType == null || fluidContainer.volume < fluidContainer.maxVolume) {
             getLiquidInReach(event.getInstigatorLocation(), event.getDirection(), 3).ifPresent(block -> {
                 EntityRef owner = item.getOwner();
+                EntityRef owner2 = owner.getOwner();
                 final EntityRef removedItem = inventoryManager.removeItem(owner, event.getInstigator(), item, false, 1);
                 //TODO: replace with better fluid handling, maybe by new CoreFluids module
+                FluidContainerItemComponent removedFluidContainer= removedItem.getComponent(FluidContainerItemComponent.class);
+                if(removedFluidContainer.fillingAmount == 0) {                                // not defined
```

Comment:
Comparing floating point numbers against a specific value may cause problems (it works for 0 and 1, but in principle there are numbers that cannot exactly be represented as floats).

Thus, a check against `<= 0` may be a bit better here.